backend/src/core/services/stats.service.ts

```diff
@@ -84,7 +86,25 @@ const getTotalSentCount = async (): Promise<number> => {
   return Statistic.sum('sent')
 }
 
+/**
+ * Helper method to get error_code, recipients for failed messages from logs table
+ * @param campaignId
+ * @param logsTable
+ */
+const getInvalidRecipients = async (campaignId: number, logsTable: string): Promise<Array<CampaignInvalidRecipient> | undefined> => {
+  // Get read replica instance
+  const sequelize = sequelizeLoader.getSequelizeReadReplicaInstance()
+  // Retrieve message logs with error codes from logs table
+  return await sequelize?.query(
+    `SELECT recipient, sent_at, updated_at, message_id, error_code FROM ${logsTable} \
```

Comment:
Hmmm would it be possible to replace `${logsTable}` with `:logsTable`?

Reply:
Also Im just curious if it is possible to use sequelize models instead of raw query to read replica db?

Reply:
Including `logsTable` in replacements quotes the string (cos it escapes the string from what I read), hence resulting in syntax error in the query. This seems to be an unresolved issue (see https://github.com/sequelize/sequelize/issues/4494). I also tried `Sequelize.literal(logTable)` to inject the value but that gave an invalid literal value error.

Yet to find a workaround, open for suggestions!

Reply:
From what I read, I would need to duplicate the models in order to associate them to another db. Since it's a single query I thought using a raw query would be cleaner.

Reply:
> Including `logsTable` in replacements quotes the string (cos it escapes the string from what I read), hence resulting in syntax error in the query. This seems to be an unresolved issue (see [sequelize/sequelize#4494](https://github.com/sequelize/sequelize/issues/4494)). I also tried `Sequelize.literal(logTable)` to inject the value but that gave an invalid literal value error.
> 
> Yet to find a workaround, open for suggestions!

Ah thanks for the explanation. Don't have any good suggestions for now haha

Reply:
> From what I read, I would need to duplicate the models in order to associate them to another db. Since it's a single query I thought using a raw query would be cleaner.

Hmmm ok maybe eventually when we have more queries on the read db we can consider models